bazel/version/gen_tmpl.py

```diff
@@ -0,0 +1,115 @@
+be read and updated from the files in order, meaning the last file's value will
+win. New keys are never read from these files. The file format parsed is chose
+to be compatible with Bazel's status file format, but any files with lines
+matching the regex `^(?P<key>[_A-Za-z][_A-Za-z0-9]) (?P<value>.*)$` may be used.
+To assist with using Bazel status files, if the key parsed from the file begins
+with `STABLE_`, that prefix is removed. Any keys which are present in the
+substitutions provided on the command line will have their value updated with
+the string read from the file.
+"""
+
+__copyright__ = """
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""
+
+import argparse
+import re
+import sys
+from pathlib import Path
+from string import Template
+
+
+_VAR_RE = re.compile(r"(STABLE_)?(?P<key>[_A-Za-z][_A-Za-z0-9]*) (?P<value>.*)")
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(__doc__)
+    parser.add_argument(
+        "--template",
+        metavar="FILE",
+        type=Path,
+        required=True,
+        help="The template source file to use.",
+    )
+    parser.add_argument(
+        "--output",
+        metavar="FILE",
+        type=Path,
+        required=True,
+        help="The output source file to produce.",
+    )
+    parser.add_argument(
+        "--substitution",
+        metavar="KEY=DEFAULT-VALUE",
+        action="append",
+        help="A substitution that should be supported and its default value.",
+    )
+    parser.add_argument(
+        "--status-file",
+        metavar="FILE",
+        type=Path,
+        action="append",
+        help="A file of key/value updates in Bazel's status file format.",
+    )
+    parser.add_argument("-v", "--verbose", action="store_true")
+    args = parser.parse_args()
+
+    # Collect the supported substitutions from the command line.
+    substitutions = {}
+    for substitution_arg in args.substitution:
+        key, value = substitution_arg.split("=", maxsplit=1)
+        substitutions.update({key: value})
+
+    # Read either of the two status files provided to build up substitutions,
+    # with the stable file last so its values override any duplicates.
+    for status_file in args.status_file if args.status_file else []:
+        if args.verbose:
+            print(f"Reading status file: {status_file}", file=sys.stderr)
+        for line in status_file.open():
+            # Remove line endings.
+            line = line.rstrip("\r\n")
+            # Exactly matches our pattern
+            if m := re.fullmatch(_VAR_RE, line):
+                key = m.group("key")
+                if key in substitutions:
+                    value = m.group("value")
+                    if args.verbose:
+                        print(f"Parsed: '{key}': '{value}'", file=sys.stderr)
+                    substitutions.update({key: value})
```

Comment:
What's actually in the status file? I'm wondering if this could be done without the regex:

```suggestion
            (key, value) = line.split(' ', 1)
            key = key.removeprefix("STABLE_")
            if key in substitutions:
                if args.verbose:
                    print(f"Parsed: '{key}': '{value}'", file=sys.stderr)
                substitutions.update({key: value})
```